sdks/python/apache_beam/runners/worker/sideinputs.py

```diff
@@ -78,29 +82,51 @@ def _start_reader_threads(self):
       t.start()
       self.reader_threads.append(t)
 
+  def _get_source_position(self, range_tracker=None, reader=None):
```

Comment:
<!--new_thread; commit:6e2c396b4cdcce20867c583cba65c8664ff79996; resolved:0-->
when does this a get range_tracker vs reader ? Please clarify with a comment.

Reply:
+1. Is there a way to avoid this either/or altogether?